x-pack/plugins/security_solution/public/detections/components/alerts_table/alerts_sub_grouping.tsx

```diff
@@ -193,17 +193,22 @@ export const GroupedSubLevelComponent: React.FC<AlertsTableComponentProps> = ({
     skip: isNoneGroup([selectedGroup]),
   });
 
-  const [queriedGroup, setQueriedGroup] = useState('');
+  const queriedGroup = useRef<string | null>(null);
 
   const aggs = useMemo(
     // queriedGroup because `selectedGroup` updates before the query response
-    () => parseGroupingQuery(queriedGroup, alertsGroupsData?.aggregations),
-    [alertsGroupsData?.aggregations, queriedGroup]
+    () =>
+      parseGroupingQuery(
+        // fallback to selectedGroup if queriedGroup.current is null, this happens in tests
+        queriedGroup.current === null ? selectedGroup : queriedGroup.current,
+        alertsGroupsData?.aggregations
+      ),
+    [alertsGroupsData?.aggregations, selectedGroup]
   );
 
   useEffect(() => {
     if (!isNoneGroup([selectedGroup])) {
-      setQueriedGroup(queryGroups?.aggs?.groupsCount?.cardinality?.field ?? '');
+      queriedGroup.current = queryGroups?.aggs?.groupsCount?.cardinality?.field ?? '';
```

Comment:
instead of duplicating this piece of state as useState or useRef, you could return `{ loading, query, ...alerts}` as part of useQuery on https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/public/detections/containers/detection_engine/alerts/use_query.tsx#L148

Reply:
the query updates before the response, so we would have the same bug.